### DISCUSSION TITLE
Removing a deprecated entry from Thor ToolBar

### DISCUSSION THREAD

**Discussion by Steingo** [VFPX/Thor#256] [Q&A] [answered]
I somehow ended up with 2 GoFish icons on my Thor toolbar - one points to the current GoFish, the other one still points to the old GoFish5 which I've removed from the Thor repository.  How can I get it off the toolbar?

**Comment by Jimrnelson** [comment]
@Steingo 

Stein -- right-click on the one you want to disappear.  One of the options should allow you to remove it from the toolbar.

Not sure if it takes immediate effect, though, you might have to run Thor again.

**Reply by Steingo** [reply]
Unfortunately that particular icon does not generate a right-click menu

**Comment by rschummer** [comment]
Hi Stein, Just in case, did you remove the GoFish 5.0 folder or did you remove the Thor tool (prg) that runs to add GF5 to the menu and toolbar? You have to remove the Thor Tool or it remains on the menu. RickPlease excuse brevity, sent from phoneOn Apr 26, 2024, at 7:52 PM, Stein Goering ***@***.***> wrote:﻿
I somehow ended up with 2 GoFish icons on my Thor toolbar - one points to the current GoFish, the other one still points to the old GoFish5 which I've removed from the Thor repository.  How can I get it off the toolbar?

**Reply by Steingo** [reply]
Don't see a GF5 folder and all the PRGs with GF5 in the name are in the Recycle folder.

**Comment by Jimrnelson** [comment]
@Steingo

One of the things never addressed in Thor was how to remove a tool.  Turns out it's messy.  

There are at least five different areas to address, and they are independent:

1 - The folder in Thor's App directory.  Removing this folder causes Thor's Check For Updates to see that it is not installed.

2 - The PRG (which is the actual way the tool is invoked).  Removing the prg causes it be removed from the 'Thor Tools' menu pad in the system menu.  

Finding the prg can be a little tricky though.  I suggest that you use Thor Tools to navigate to the tool, shift-click on the tool and choose the `'View Published Version (Read-Only)'` menu item.  This puts the name of the prg in your MRU list.  But don't erase it yet.
 
3 - Whether the tool appears in Thor's tool bar.  If is in the tool bar, you can remove it by using one of the options in the menu described in the option above.
 
4 - Whether the tool is assigned a hot key.  If is assigned a hot key, you can remove it by using one of the options in the menu described in the option above.

5 -  Whether appears in any user defined menus.  You'll have to modify those menus manually use Thor configuration form.

If you prefer to live dangerously and fiddle with the Thor tables directly, look at the DBFs in `_screen.cThorFolder + 'Tables'`.  The PRG name (without path, but with extension) is uses as the key in a number of tables there.

**Reply by Steingo** [reply]
Thanks.  I'll look into these suggestions when I get a chance.

**Reply by Steingo** [reply, marked as answer]
I managed to clean up most of those references but never found a menu option for tool bar removal.  Finally got the icon to disappear by browsing the ToolBarTools table to the GoFish5 record and setting Enabled to F

**Comment by Jimrnelson** [comment]
@Steingo 

Unfortunately, at no time during the development of Thor was any consideration given to tools or apps could be deleted.  And questions like yours have come up so infrequently that nothing has ever been done about it.

Glad to see you got this taken care of.

**Comment by Steingo** [comment]
The process was a bit involved but for the most part pretty straightforward.  Where I went wrong was leaving the Toolbar icon for the last.  I should have, at the outset, used the right-click toolbar menu to unclick the Enable box, but that menu disappeared after I zapped the other GF5 components, which meant I had to edit the table directly.

**Comment by Jimrnelson** [comment]
Yup, a messy process.  Sorry about that.  Not sure what to do with that going forward ...

**Reply by Steingo** [reply]
Not sure it was messy enough to justify trying to automate the removal process.  Maybe a help screen somewhere that outlines the process and makes the point about clearing the tool bar as the first step.

**Comment by Jimrnelson** [comment]
@Steingo 

> Not sure it was messy enough to justify trying to automate the removal process. Maybe a help screen somewhere that outlines the process and makes the point about clearing the tool bar as the first step.

Good point.  Can you add this on the Issues page (same text) so that it will not get lost?  Thanks / jim
